Contents/Code/__init__.py

```diff
@@ -111,162 +125,134 @@ def update(self, metadata, media, lang):
+				person_info_role = person_info_role[0]
+
+			person_profession = person_list.xpath(".//div[@class = 'personRole roleSource ']/@data-profession")
+
+			if person_profession is not None and len(person_profession) > 0:
+				if person_profession[0] == 'actors':
+					role = metadata.roles.new()
+					role.role = person_info_role
+					role.name = person_info_name
+					role.photo = person_image
+
+		# DIRECTORS, WRITERS
+		crew_url = "{}{}/cast/crew".format(FILMWEB, movie_id)
+		info_filmweb_crew_result = HTML.ElementFromURL(crew_url)
+
+		metadata.directors.clear()
 		metadata.writers.clear()
-		writers_links = info_filmweb_result.xpath(".//div[contains(@class, 'filmInfo')]//th[text()=starts-with(.,'sce')]/parent::tr//a/@href")
-		if len(writers_links) > 0 and str(writers_links[-1]).startswith('/film'):
-			# get writers from crew page
-			if crew_page is None:
-				crew_url = FILMWEB + str(writers_links[-1]).replace('#screenwriter', '')
-				info_filmweb_crew_result = HTML.ElementFromURL(crew_url, 0, {'Cookie':HTTP.CookiesForURL(FILMWEB)}, cacheTime = 0)
+		for person_element in info_filmweb_crew_result.xpath("//div[@class = 'personRole roleSource ']"):
+			if person_element.get('data-profession') == 'director':
+				metadata.directors.new().name = person_element.get('data-person')
+			elif person_element.get('data-profession') == 'screenwriter':
+				metadata.writers.new().name = person_element.get('data-person')
 
-			for writer in info_filmweb_crew_result.xpath("//form/h2[starts-with(text(), 'sce')]/following-sibling::div[1]/table/tr/td/a[starts-with(@href, '/person') and @title]/@title"):
-				metadata.writers.new().name = writer.strip()
-				Log("WRITER: " + writer.strip())
-			pass
-		else:
-			for writers in info_filmweb_result.xpath(".//div[contains(@class, 'filmInfo')]//th[text()=starts-with(.,'sce')]/parent::tr//a"):
-				try:
-					writer = writers.text_content().strip()
-					metadata.writers.new().name = writer
-					Log("WRITER: " + writer)
-				except:
-					pass		
+		# MAIN POSTER
+		# TODO: setPoster("http://1.fwcdn.pl/po/19/57/731957/7751671.0.jpg")
 
-		#POSTERS
+		# OTHER POSTERS
 		info_filmweb_posters_result = HTML.ElementFromURL(FILMWEB + movie_id + '/posters', 0, {'Cookie':HTTP.CookiesForURL(FILMWEB)}, cacheTime = 0)
     
 		#del metadata.posters
 		if Prefs['delthumb']:
 			for x in range(len(metadata.posters)):
 				for item in metadata.posters:
 					t = item
-					break;
+					break
 				del metadata.posters[t]  
          
-		num_posters = Prefs['numthumb'];
-		i = 0;
+		num_posters = Prefs['numthumb']
+		i = 0
 		sort = 1
-		for posters in info_filmweb_posters_result.xpath(".//ul[contains(@class, 'postersList')]/li"):
+		for posters in info_filmweb_posters_result.xpath(".//img[@class = 'simplePoster__image']/@data-src"):
 			try:
-				poster_url = posters.xpath(".//img")[0].get('src').replace('2.jpg','3.jpg')
+				poster_url = posters.replace('2.jpg', '3.jpg')
```

Comment:
Zdjęcie plakatu kończy się teraz na 6.jpg, więc: poster_url = posters.replace('6.jpg', '3.jpg')

Reply:
```suggestion
				poster_url = posters.replace('6.jpg', '3.jpg')
```

Reply:
Faktycznie przeoczyłem. Pobierane były, tylko w mniejszej wersji :)

Reply:
> Wow :) Pierwsze co to dziękuję za mega robotę :)
> 
> Obecnie przebywam poza miejscem zamieszkania i nie mam jak przetestować, więc tylko wizualnie sprawdziłem z kodem html strony filmweb.pl i dwie rzeczy zauważyłem do poprawienia przed zatwierdzeniem (chyba że się mylę, coś przeoczyłem).

W zasadzie to ja dziękuję za stworzenie pierwszej wersji i jej udostępnienie. To była największa praca w to włożona ;) 
Tak na marginesie, to pominąłem pobieranie ciekawostek, bo nie znalazłem miejsca, żeby się gdziekolwiek pokazywały.